packages/css-variables-language-server/src/CSSVariableManager.ts

```diff
@@ -66,6 +65,54 @@ const getAST = (filePath: string, content: string) => {
 export default class CSSVariableManager {
   private cacheManager = new CacheManager<CSSVariable>();
 
+  private resolveCachedVariables = () => {
+    for (const filePath of this.cacheManager.getFiles()) {
+      this.cacheManager.getAll(filePath).forEach((variable, key) => {
+        this.setCssVariable(key, this.resolveRecursiveVariables(variable.symbol.value), filePath, variable.definition.range);
+      });
+    }
+  };
+
+  public resolveRecursiveVariables = (value: string) => {
+    for (let i = 0; i < 20; i++) {
+      const variableReference = value.match(/^var\(\s*([a-zA-Z0-9-]+)\s*\)$/)?.[1];
+      if (variableReference) {
+        const newValue = this.cacheManager.get(variableReference)?.symbol?.value;
+        if (newValue) {
+          value = newValue;
+        } else {
+          break;
+        }
+      } else {
+        break;
+      }
```

Comment:
```suggestion
        const variableReference = value.match(/^var\(\s*([a-zA-Z0-9-]+)\s*\)$/)?.[1];
        if (!variableReference) {
            break;
        }
        const newValue = this.cacheManager.get(variableReference)?.symbol?.value;

        if (newValue) {
          value = newValue;
          continue;
        }

        break;
```

I think nested if statements are complicated. How about this?